packages/loader/container-loader/src/runtimeCompatManager.ts

```diff
@@ -0,0 +1,68 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	LayerCompatibilityManager,
+	type ICompatibilityDetails,
+} from "@fluid-internal/client-utils";
+import type { ICriticalContainerError } from "@fluidframework/container-definitions";
+import { UsageError } from "@fluidframework/telemetry-utils/internal";
+
+import { pkgVersion } from "./packageVersion.js";
+
+/**
+ * This class manages the state and validation to ensure that the Loader layer is compatible with the Runtime layer.
+ * @internal
+ */
+export class RuntimeLayerCompatManager extends LayerCompatibilityManager {
```

Comment:
i generally prefer composition over inheritance, unless there is a need for  Liskov substitution. I find inheritance just makes the code brittle, and is harder to test. in this case composability would be better, so just use a LayerCompatibilityManager, rather than extend it.

Reply:
see my comments on layerCompat.ts

Reply:
I changed this to implement `ICompatibilityDetails` now.
I kept all the data in this class to not add more things to the `Container` class (same for the one in container runtime). It also makes unit testing these easier.